CMakeLists.txt

```diff
@@ -19,27 +23,60 @@ set(SOURCES
     src/loop/loop.cpp
     src/track/track.cpp
     src/ui/ui.cpp
+    mainwindow.cpp
+    mainwindow.ui
 )
 
 # Add the include directory, this is for our 3rd party libraries we will use
 include_directories(include)
 
 # Find required packages
 find_package(SQLite3 REQUIRED)
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Widgets)
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Widgets)
 
 # Add executable
-add_executable(audio-loop-station ${SOURCES})
+if(${QT_VERSION_MAJOR} GREATER_EQUAL 6)
+    qt_add_executable(audio_station_loop
+        MANUAL_FINALIZATION
+        ${SOURCES}
+    )
+else()
+    add_executable(audio_station_loop
+        ${SOURCES}
+    )
+endif()
+# add_executable(audio-loop-station ${SOURCES})
 
 # Link Libraries. This is where boost, rtaudio, etc will go when ready
-target_link_libraries(audio-loop-station PRIVATE
+target_link_libraries(audio_station_loop PRIVATE
     SQLite::SQLite3
     sqlite3
+    Qt${QT_VERSION_MAJOR}::Widgets
 )
 
 # Set the output directory
-set_target_properties(audio-loop-station PROPERTIES 
-    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/build")
+#set_target_properties(audio_station_loop PROPERTIES
+#    RUNTIME_OUTPUT_DIRECTORY "${CMAKE_SOURCE_DIR}/build")
+
+set_target_properties(audio_station_loop PROPERTIES
+    MACOSX_BUNDLE_GUI_IDENTIFIER my.example.com
+    MACOSX_BUNDLE_BUNDLE_VERSION ${PROJECT_VERSION}
+    MACOSX_BUNDLE_SHORT_VERSION_STRING ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR}
+    MACOSX_BUNDLE TRUE
+    WIN32_EXECUTABLE TRUE
+)
```

Comment:
With us only developing on linux, what do these properties do? 

Also, with the build property deleted, will we still be able to build within a build directory?